internal/plan/plan.go

```diff
@@ -40,15 +44,28 @@ type OptimisationValue struct {
 	Unit  int     `json:"unit"`
 }
 
-func (o OptimisationPlan) IsEmpty(logger *slog.Logger) bool {
+func (o OptimisationPlan) IsEmpty() bool {
 	return o.SiteID == "" && len(o.OptimisationIntervals) == 0 && o.OptimisationTimestamp.Seconds == 0
 }
 
-func NewHandler(path string) PlanHandler {
-	return PlanHandler{Path: path}
+func (i OptimisationInterval) IsEmpty() bool {
+	return i.Interval.StartTime.Seconds == 0
 }
 
-func (p PlanHandler) ReadPlan() (OptimisationPlan, error) {
+func (i OptimisationInterval) IsCurrent(targetTime time.Time) bool {
+	intStart := time.Unix(i.Interval.StartTime.Seconds, 0)
+	intEnd := time.Unix(i.Interval.EndTime.Seconds, 0)
+
+	isAfterStart := targetTime.Equal(intStart) || targetTime.After(intStart)
+	isBeforeEnd := targetTime.Before(intEnd)
+	return isAfterStart && isBeforeEnd
+}
+
+func NewHandler(logger *slog.Logger, path string) Handler {
+	return Handler{logger: logger, mu: new(sync.RWMutex), Path: path}
+}
+
+func (p Handler) ReadPlan() (OptimisationPlan, error) {
```

Comment:
should this use the `RLock` method of the mutex?

Reply:
yes it should, thanks for pointing that out!